### ISSUE TITLE
Feature Request: More renaming varibles (For Anime)

### ISSUE THREAD

**Issue by joakimlemb** [pymedusa/Medusa#4451]
I'm trying to create a rename rules that match Anime releases, but it does not seem possible with the current available variables in Medusa, so this feature request is about adding a few new variables so the following rename rule might work:

- A variable for CRC32 file hash (%CRC32)
- A variable for only the resolution (%RES), currently resolution and source is merged into %QN.
- A variable for only the source (%SRC), currently resolution and source is merged into %QN.
- A varible for anime versioning if it is not handled by %RT for v1, v2 etc...

A example of the rename rule I'm trying to create:
[%RG]_%S_N_-_%0E_[%RT][%RES][%SRC][%CR32].ext

which for example should result in something like: 
[GROUP-NAME]_SERIES_NAME_-_03_[v2][1080p][FFFFFFFF].mkv

Or if any third party scripts for Medusa is available for this?
(Or it would be even better if we could use all the data available in the Anidb API for renaming)

**Comment by hergonoway** [comment]
I think it's a very good idea. 

More, being able to hash the file could be use to enhance the synchronization with anidb account to the next level. I'm currently post processing files from Medusa to another folder with their "aniadd" app just because their database is really solid (even if its a manual process I do 2/3 times a month), and could solve many problems caused by anime's "way of doing stuff with season, renaming Anime mid-season, and prequel which is not anymore and OAV, etc".  With the hash and their current API, Medusa should be able to enhance scene name detection and synchronisation with anidb account.

**Comment by p0psicles** [comment]
Im trying the understand the reason behind wanting to recreate the release name. Why not just leave the original name intact?

**Comment by joakimlemb** [comment]
@p0psicles , To fix those that deviate from the standard.

**Comment by p0psicles** [comment]
Okay, there isn't really a standard in anime naming afaik. But sure I understand the desire to have them "standardized" using your own format.

Other question, the CRC. Should that hash of the 'old/original' filename? Or of the recreated filename?
Also how is that used to identify them on anidb? Do you have some docs on that?

**Comment by joakimlemb** [comment]
True, there probably isn't a documented and agreed upon standard, yet most releases follows the same pattern with some exceptions.

The CRC32 hash is of the file content, not the filename. 
Anidb has a lot of API documentation on their wiki, here is the link for file related stuff: 
https://wiki.anidb.net/w/UDP_API_Definition#FILE:_Retrieve_File_Data
and:
https://wiki.anidb.net/w/Content:Files

**Comment by p0psicles** [comment]
Okay I got it.
So what we need to do for this is:
* Add a new field to tv_episodes for storing the crc
* calculate and save the crc on each postprocess.
* modify Qulality class for returning separated resolution and source
* adapt postprocessing. Will depend on #4259 

This one: `A varible for anime versioning if it is not handled by %RT for v1, v2 etc...` is not really clear to me.

**Comment by joakimlemb** [comment]
In the same way that The Scene uses PROPER and REPACK tags in filenames, Anime releases uses a versioning system: https://wiki.anidb.net/w/Content:Files#Version

Releases which is "Version 1/Initial Release" does not have this tag in the filename, but if for example a PROPER is needed for an episode, that episode is tagged with [v2] to indicate that episode has fixed some kind of issue which existed within a v1 release.

**Comment by joakimlemb** [comment]
There might be a lot of timed saved if maybe Medusa just integrated the following python Anidb UDP API client: https://github.com/adameste/anidbcli
